sw/device/silicon_creator/manuf/skus/earlgrey_a0/generic/sram_cp_provision_init.c

```diff
@@ -61,40 +62,134 @@ static status_t peripheral_handles_init(void) {
-      kFlashInfoFieldDeviceId, kFlashInfoFieldManufState,
-      kFlashInfoFieldWaferAuthSecret};
-  const uint32_t *data[kNumFlashInfoFields] = {
-      provisioning_data->device_id, provisioning_data->manuf_state,
-      provisioning_data->wafer_auth_secret};
-  const uint32_t data_sizes[kNumFlashInfoFields] = {
-      kHwCfgDeviceIdSizeIn32BitWords, kHwCfgManufStateSizeIn32BitWords,
-      kFlashInfoWaferAuthSecretSizeIn32BitWords};
-  uint32_t byte_address;
-
-  for (size_t i = 0; i < ARRAYSIZE(fields); ++i) {
-    byte_address = 0;
-    TRY(flash_ctrl_testutils_info_region_setup(
-        &flash_ctrl_state, fields[i].page, fields[i].bank, fields[i].partition,
-        &byte_address));
-    // We skip the erasing of the ManufState (at index 1 in the arrays above)
-    // field page as this field shares a page with the DeviceId field.
-    if (i == 1) {
-      TRY(flash_ctrl_testutils_write(
-          &flash_ctrl_state, byte_address, fields[i].partition, data[i],
-          kDifFlashCtrlPartitionTypeInfo, data_sizes[i]));
-    } else {
-      TRY(flash_ctrl_testutils_erase_and_write_page(
-          &flash_ctrl_state, byte_address, fields[i].partition, data[i],
-          kDifFlashCtrlPartitionTypeInfo, data_sizes[i]));
-    }
-  }
+  uint32_t byte_address = 0;
+  // DeviceId and ManufState are located on the same flash info page.
+  TRY(flash_ctrl_testutils_info_region_setup(
+      &flash_ctrl_state, kFlashInfoFieldDeviceId.page,
+      kFlashInfoFieldDeviceId.bank, kFlashInfoFieldDeviceId.partition,
+      &byte_address));
+  TRY(flash_ctrl_testutils_erase_page(&flash_ctrl_state, byte_address,
+                                      kFlashInfoFieldDeviceId.partition,
+                                      kDifFlashCtrlPartitionTypeInfo));
+  return OK_STATUS();
+}
+
+static status_t write_device_id_and_manuf_state_flash_info_page(
+    manuf_cp_provisioning_data_t *provisioning_data) {
+  // Write DeviceId.
+  uint32_t byte_address = 0;
+  TRY(flash_ctrl_testutils_info_region_setup(
+      &flash_ctrl_state, kFlashInfoFieldDeviceId.page,
+      kFlashInfoFieldDeviceId.bank, kFlashInfoFieldDeviceId.partition,
+      &byte_address));
+  TRY(flash_ctrl_testutils_write(
+      &flash_ctrl_state, byte_address, kFlashInfoFieldDeviceId.partition,
+      provisioning_data->device_id, kDifFlashCtrlPartitionTypeInfo,
+      kHwCfgDeviceIdSizeIn32BitWords));
+
+  // Write ManufState.
+  byte_address = 0;
+  TRY(flash_ctrl_testutils_info_region_setup(
+      &flash_ctrl_state, kFlashInfoFieldManufState.page,
+      kFlashInfoFieldManufState.bank, kFlashInfoFieldManufState.partition,
+      &byte_address));
+  TRY(flash_ctrl_testutils_write(
+      &flash_ctrl_state, byte_address, kFlashInfoFieldManufState.partition,
+      provisioning_data->manuf_state, kDifFlashCtrlPartitionTypeInfo,
+      kHwCfgManufStateSizeIn32BitWords));
 
   return OK_STATUS();
 }
 
+static status_t erase_wafer_auth_secret_flash_info_page(
```

Comment:
nit: verb as a suffix to maintain consistency with other silicon_creator code... (even though we are not super consistent).

Reply:
fixed.